detect_os

```diff
@@ -45,11 +45,17 @@ usage()
 
     echo -e "\t -h/--help/--usage"
     echo -e "\t\t Displays this message"
+}
 
-    exit 0
+install_deps()
+{
+    if [ ! which yq > /dev/null 2>&1 ]; then
+        pip3 install yq
```

Comment:
We will stop here, as it might have a ripple effect.
What do you do if the system does not have pip/pip3 installed.  IE, an AWS rhel image
[root@ip-172-31-46-204 ec2-user]# which pip3
/usr/bin/which: no pip3 in (/root/.local/bin:/root/bin:/sbin:/bin:/usr/sbin:/usr/bin)

Reply:
Now fetching from prebuilt binaries.